rohmu/delta/snapshot.py

```diff
@@ -177,13 +178,22 @@ def _snapshot_create_missing_directories(self, *, src_dirs: Sequence[Path], dst_
             changes += 1
         return changes
 
-    def _snapshot_remove_extra_files(self, *, src_files: Sequence[Path], dst_files: Sequence[Path]) -> int:
+    def _snapshot_remove_extra_files(
+        self,
+        *,
+        src_files: Sequence[Path],
+        dst_files: Sequence[Path],
+        progress: Progress,
```

Comment:
if `progress` is required, this will be a backwards-compatible change. It may be better to have a(n immutable) default value here.

Reply:
yep, I made the progress parameter optional and default to None. Inside the method, if progress isn't provided, a new Progress instance is created, ensuring backward compatibility and optional progress tracking.